common/zeroconf/zeroconf.h

```diff
@@ -0,0 +1,210 @@
+		virtual Target& insert(const std::string & name, const int & port);
+		virtual Target& insert(const Dashel::Stream* dashel_stream);
+
+		//! Aseba::Zeroconf can update its knowledge of non-local targets by browsing the network
+		virtual void browse();
+
+	protected:
+		std::vector<Target> targets; //!< the targets in this container
+		void registerTarget(Target * target, const TxtRecord& txtrec); //!< requested through target
+		void updateTarget(const Target * target, const TxtRecord& txtrec); //!< requested through target
+		void resolveTarget(Target * target); //!< requested through target
+
+	public:
+		//! An error in registering or browsing Zeroconf
+		struct Error:public std::runtime_error
+		{
+			Error(const std::string& what): std::runtime_error(what) {}
+		};
+
+	protected:
+		//! Private class that encapsulates an active service request to the mDNS-SD daemon
+		class ZeroconfDiscoveryRequest
+		{
+		public:
+			DNSServiceRef serviceref{nullptr};
+			bool in_process{true}; //!< flag to signal when this request is no longer active
+		public:
+			virtual ~ZeroconfDiscoveryRequest()
+			{   // release the service reference
+				if (serviceref)
+					DNSServiceRefDeallocate(serviceref);
+				serviceref = 0;
+			}
+			virtual bool operator==(const ZeroconfDiscoveryRequest &other) const
+			{
+				return serviceref == other.serviceref;
+			}
+		};
+
+		ZeroconfDiscoveryRequest browseZDR; //! the zdr for browse requests isn't attached to a target
+
+	protected:
+		//! Called when the browsing process is complete.
+		//! Can be overridden in derived classes to schedule an update to the UI.
+		virtual void browseComplete() {}
+
+		//! The discovery request can be processed immediately, or can be registered with
+		//! an event loop for asynchronous processing.
+		//! Can be overridden in derived classes to set up asynchronous processing.
+		virtual void processDiscoveryRequest(ZeroconfDiscoveryRequest & zdr);
+
+	private:
+		//! callbacks for the DNS Service Discovery API
+		static void DNSSD_API cb_Register(DNSServiceRef sdRef, DNSServiceFlags flags, DNSServiceErrorType errorCode, const char *name, const char *regtype, const char *domain, void *context);
+		static void DNSSD_API cb_Browse(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t interfaceIndex, DNSServiceErrorType errorCode, const char *serviceName, const char *regtype, const char *replyDomain, void *context);
+		static void DNSSD_API cb_Resolve(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t interfaceIndex, DNSServiceErrorType errorCode, const char *fullname, const char *hosttarget, uint16_t port, /* In network byte order */ uint16_t txtLen, const unsigned char *txtRecord, void *context);
+	};
+
+	/**
+	 \addtogroup zeroconf
+
+		A Zeroconf::Target allows client classes to choose and access Aseba targets.
+	 */
+	class Zeroconf::Target
+	{
+	public:
+		std::string name;
+		int port;
+		std::string domain{"local."};
```

Comment:
We could also use the `=` syntax, as it would match the initialization of static fields in C++98, but `{}` is also fine, as it denotes a C++11 syntax. Just we should write this choice somewhere so that we are consistent.